prost-derive/src/field/map.rs

```diff
@@ -275,17 +275,11 @@ impl Field {
             Some(quote! {
                 #[doc=#get_doc]
                 pub fn #get(&self, key: #key_ref_ty) -> ::core::option::Option<#ty> {
-                    self.#ident.get(#take_ref key).cloned().and_then(|x| {
-                        let result: ::core::result::Result<#ty, _> = ::core::convert::TryFrom::try_from(x);
-                        result.ok()
-                    })
+                    self.#ident.get(#take_ref key).cloned().and_then(|x| { x.known() })
                 }
                 #[doc=#insert_doc]
                 pub fn #insert(&mut self, key: #key_ty, value: #ty) -> ::core::option::Option<#ty> {
-                    self.#ident.insert(key, value as i32).and_then(|x| {
-                        let result: ::core::result::Result<#ty, _> = ::core::convert::TryFrom::try_from(x);
-                        result.ok()
-                    })
+                    self.#ident.insert(key, value.into()).and_then(|x| { x.known() })
```

Comment:
Is it beneficial to provide a `get` and `insert` function with the new wrapper? I feel like the map can be used directly with the helper functions provided by OpenEnum.

Reply:
As can be seen in the generated code itself, these methods provide considerable convenience for the most common use cases.

Reply:
Perhaps the insert method could be changed to return the possible old value as `Option<OpenEnum>` so the caller can choose the fallback behavior for unknown values. So yes, the only added convenience there would be the `.into()` conversion call.